src/components/button-favourite/button-favourite.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { useEffect, useState } from "react";
+import { IconFavourite } from "../icon-favourite/icon-favourite";
+import {
+  addItem,
+  removeItem,
+  useHasItem
+} from "../../core/material-list-api/material-list";
+
+export interface ButtonFavouriteProps {
+  materialId: string;
+}
+
+// TODO We have to check if user is login and redirect if not
+
+const ButtonFavourite: React.FC<ButtonFavouriteProps> = ({ materialId }) => {
+  const [fillState, setFillState] = useState<boolean>(false);
+  const { mutate } = useHasItem();
+
+  useEffect(() => {
+    mutate(
+      {
+        listId: "default",
+        itemId: materialId
+      },
+      {
+        onSuccess: () => {
+          setFillState(true);
+        },
+        // The material list service will return response code 404 when a
+        // material is not on the patrons list. This is interpreted as an
+        // error by our client. Consequently we set
+        onError: () => {
+          setFillState(false);
+        }
+      }
+    );
+  }, [materialId, mutate]);
+
+  const handleClick = () => {
+    if (fillState) {
+      removeItem("default", materialId);
+      setFillState(false);
+    } else {
+      addItem("default", materialId);
+      setFillState(true);
+    }
+  };
+
+  return (
+    <button
+      type="button"
+      aria-label={fillState ? "Fjern fra favoritter" : "Tilføj til favoritter"}
```

Comment:
Texts as props?

Reply:
I'm not sure what you mean? fillState is a boolean that determines which of the texts is displayed?

Reply:
I mean, the aria-label should be a prop?

Reply:
Okay, we have switched to using useText() instead, please see this commit:
https://github.com/danskernesdigitalebibliotek/dpl-react/pull/43/commits/50f5237532b03cd701ae1b183a039072eb363839#diff-8eccef2c16d12483231c47763961a184babc8f6788f6e3883890726714ccb3ca

Reply:
But all tekst should come from the CMS.